Generate swagger common parameters with Jax-RS annotations

cjelger opened the pull request.

cjelger commented:
@Postremus Would you have to review this PR? Thanks in advance.

Postremus commented:
@cjelger I will take a look this evening (CEST). However, please keep in mind that I am also no maintainer of this projects, and therefore can't accept any pull requests.

cjelger commented:
ok thx ... who are currently the maintainers of the project?

Postremus commented:
*who* and *kongchen* are, as far as I can see, the maintainers of this projects.

Postremus commented on src/main/java/com/github/kongchen/swagger/docgen/reader/JaxrsReader.java:
```diff
@@ -194,6 +198,24 @@ public String value(){
     }
 
+    private void readCommonParameters(Class<?> cls) {
+        Path path = AnnotationUtils.findAnnotation(cls, Path.class);
```
> One possibility would be to annotate a Class with io.swagger.annotations.Api but without any javax.ws.rs.Path annotation, to indicate that this class is not specific to any path and only contains "common/global" parameters.

I believe that we should also look on methods for a Path annotation to determine if a given class is resource class.
Also, we should check if the method is annotated with any of the HTTP Method Verbs (*PUT*, *GET*, *POST*, etc..).
Compare this with the document *JSR-000370 JAX-RS 2.1 Final Release Specification for Evaluation*, section 3.1:

>Resource classes are POJOs that have at least one method annotated with@Path or a request method desig-
nator.



See method extractOperationMethod in JaxrsReader.

cjelger commented:
@Postremus Thank you very much for your review. I addressed all your comments and also added an extra JSON example file and a unit test, to better document and test the feature.

Postremus commented on src/main/java/com/github/kongchen/swagger/docgen/reader/JaxrsReader.java:
```diff
@@ -194,6 +198,37 @@ public String value(){
+            if (annotations.length > 0) {
+                List<Parameter> params = getParameters(cls, Arrays.asList(annotations));
+                for (Parameter param : params) {
+                    swagger.addParameter(param.getName(), param);
```
Currently, if a parameter already exists with the same name, it is simply overwritten. This could lead to unexpected behaviour, since it is unclear which parameter will be in the generated swagger.json in the end.
We may want to stop the build at this point and inform the user about this.
What do you think?

who commented:
@cjelger This looks promising. Please resolve conflicts and re-submit so I can merge.

cjelger commented:
@who I am on PTO until Sept. 17 so I will take care of the conflicts when I'm back, thx!

cjelger commented:
Closes #640
